clang/lib/CIR/Dialect/IR/CIRDialect.cpp

```diff
@@ -1534,14 +1549,21 @@ void cir::FuncOp::print(OpAsmPrinter &p) {
       p, *this,
       {getSymVisibilityAttrName(), getAliaseeAttrName(),
        getFunctionTypeAttrName(), getLinkageAttrName(), getBuiltinAttrName(),
-       getNoProtoAttrName()});
+       getNoProtoAttrName(), getExtraAttrsAttrName()});
+
 
   if (auto aliaseeName = getAliasee()) {
     p << " alias(";
     p.printSymbolName(*aliaseeName);
     p << ")";
   }
 
+  if (!getExtraAttrs().getElements().empty()) {
+    p << " extra(";
+    p.printOptionalAttrDict(getExtraAttrs().getElements().getValue());
```

Comment:
Is the dictionary optional? That's not my understanding

Reply:
The dictionary is not optional. It's named like that because it won't print anything for an empty dictionay:

https://github.com/llvm/clangir/blob/c1fa60b4cde512964544ab66404dea79dbc5dcb4/mlir/lib/IR/AsmPrinter.cpp#L726